File: Backend/src/main/java/com/luckyseven/backend/domain/settlements/dto/SettlementUpdateRequest.java
package com.luckyseven.backend.domain.settlements.dto;

import com.luckyseven.backend.domain.settlements.entity.Settlement;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Digits;
import java.math.BigDecimal;
import lombok.Builder;

/**
 * DTO for {@link Settlement}
 */
@Builder
public record SettlementUpdateRequest(
    @DecimalMin(value = "0.0", inclusive = false)
    @Digits(integer = 15, fraction = 2)
    BigDecimal amount,

    Boolean isSettled,

    Long settlerId,

    Long payerId,

    Long expenseId) {

}

File: Backend/src/main/java/com/luckyseven/backend/domain/settlements/dto/SettlementUpdateRequest.kt
package com.luckyseven.backend.domain.settlements.dto

import jakarta.validation.constraints.NotNull
import java.math.BigDecimal

data class SettlementUpdateRequest(
    @field:NotNull
    val settlementId: Long,

    val amount: BigDecimal? = null,
    val settlerId: Long? = null,
    val payerId: Long? = null,
    val expenseId: Long? = null,
    val isSettled: Boolean? = null
) 